src/otaclient/main.py

```diff
@@ -77,11 +81,59 @@ def _signal_handler(signal_value, _) -> None:  # pragma: no cover
     _on_shutdown(sys_exit=True)
 
 
+def _thread_dynamic_client(
```

Comment:
try to run new application three times. If all of cases are failed, restart the existing otaclient.